# Useful references to check out

## Identifying cell barcodes

- [Identify cell barcodes from single-cell genomics sequencing experiments](https://github.com/pachterlab/sircel)
- [Barcode identification for single cell genomics](http://www.biorxiv.org/content/biorxiv/early/2017/05/09/136242.full.pdf)
- [Estimating the number of true cell barcodes in single cell RNA-Seq (part 1)](https://cgatoxford.wordpress.com/2017/05/18/estimating-the-number-of-true-cell-barcodes-in-single-cell-rna-seq/)
- [Estimating the number of true cell barcodes in single cell RNA-Seq (part 2)](https://cgatoxford.wordpress.com/2017/05/23/estimating-the-number-of-true-cell-barcodes-in-single-cell-rna-seq-part-2/)

## Working with UMI

- [UMI-tools](https://github.com/CGATOxford/UMI-tools)
- [UMI-tools: Modelling sequencing errors in Unique Molecular Identifiers to improve quantification accuracy](http://genome.cshlp.org/content/early/2017/01/18/gr.209601.116.full.pdf)


## Pipeline

- [dropSeqPipe](https://github.com/Hoohm/dropSeqPipe)
- [Instructions for dropSeqPipe](https://github.com/Hoohm/dropSeqPipe/wiki)
- [Drop-seq-tools cookbook](http://mccarrolllab.com/wp-content/uploads/2016/03/Drop-seqAlignmentCookbookv1.2Jan2016.pdf)
